### PR TITLE
London8_Mandeep_Sangha_JS1_wk2

### PR DESCRIPTION
**Pull request by mandeepsangha**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

File: exercises/C-comparison-operators/exercise.js [hunk 1]
```diff
@@ -5,16 +5,16 @@
   The variables should have values that match the expected results.
 */
 
-var studentCount = 16;
-var mentorCount = 9;
-var moreStudentsThanMentors = studentCount > mentorCount; // finish this statement
+let studentCount = 16;
```
Comment (teniolao) [inline exercises/C-comparison-operators/exercise.js:8]: Good code layout Mandeep, it's great you were able to pass all the test.

**Review by teniolao — COMMENTED** [review]
Good code layout Mandeep, it's great you were able to pass all the test.

File: extra/1-radio-stations.js [hunk 2]
```diff
@@ -15,6 +15,15 @@
 
 // `getAllFrequencies` goes here
 
+function getAllFrequencies(){
+  let array = [];
+    for (i = 87; i < 109; i++) {
```
Comment (TimeaReich) [inline extra/1-radio-stations.js:20]: Hi! I am reviewing your code, and it looks like you know a lot more than me, so I don't think I can add anything, but i sure learned a few things going through your code so thank you! Nice use of for loop here.

File: exercises/G-conditionals-2/exercise-3.js [hunk 3]
```diff
@@ -8,7 +8,18 @@
   - Otherwise the grade is "F"
 */
 
-function calculateGrade(mark) {}
+function calculateGrade(mark) {
```
Comment (MatenRehimi) [inline exercises/G-conditionals-2/exercise-3.js:11]: Function works correctly but some of the conditions in the if statements are not needed. Try to have a look at which ones might not be need to simplify logic here.

File: exercises/I-array-properties/exercise.js [hunk 4]
```diff
@@ -5,8 +5,13 @@
 
 */
 
-function isEmpty(arr) {
-  return; // complete this statement
+function isEmpty(arr) { // complete this statement
+  if (arr.length === 0){
```
Comment (MatenRehimi) [inline exercises/I-array-properties/exercise.js:9]: Could replace the if-else with a predicate similar to the function below
function isNegative(number) {
  return number < 0;
}

File: mandatory/2-function-creation.js [hunk 5]
```diff
@@ -25,7 +39,20 @@ Write a function that:
 - removes any forward slashes (/) in each string
 - makes all strings all lowercase
 */
-function tidyUpStrings(arrayOfStrings) {}
+function tidyUpStrings(array) {
```
Comment (MatenRehimi) [inline mandatory/2-function-creation.js:42]: For this function, it's possible to call multiple methods when returning. So you could have el.trim().toLowerCase().replace(/\//g, '')

File: extra/1-radio-stations.js [hunk 6]
```diff
@@ -25,6 +34,11 @@
  * - Return only the frequencies that are radio stations.
  */
 // `getStations` goes here
+function getStations(){
+let array2 = getAllFrequencies().filter(x => isRadioStation(x));
```
Comment (MatenRehimi) [inline extra/1-radio-stations.js:38]: Good use of filter here to produce clean/readable code.

File: exercises/E-conditionals/exercise.js [hunk 7]
```diff
@@ -9,6 +9,14 @@
 let name = "Daniel";
 let danielsRole = "mentor";
 
+if (danielsRole = "mentor"){
```
Comment (MatenRehimi) [inline exercises/E-conditionals/exercise.js:12]: Comparisons should be ===, in this instance you've used = as assignment and given the variable danielsRole the value "mentor again. 

Be careful as danielsRole = "mentor" will still passes the tests as assignments with strings results to true if the length of the string is greater than 1. Here's an example to show this:
console.log(Boolean((names = "h")));   //true
console.log(Boolean((names = "")));    //false

**Pull request closed without merging**